src/components/PaymentStripe/cardSection.js

```diff
@@ -0,0 +1,136 @@
+import React, { useMemo, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { CardElement, useElements, useStripe } from '@stripe/react-stripe-js';
+import { useTheme, useUniqueIdentifier } from '../../libs/hooks';
+import RenderIf from '../RenderIf';
+import Label from '../Input/label';
+import HelpText from '../Input/styled/helpText';
+import ErrorText from '../Input/styled/errorText';
+import StyledContainer from './styled/container';
+
+const CardSection = props => {
+    const {
+        label,
+        hideLabel,
+        bottomHelpText,
+        error,
+        className,
+        style,
+        hidePostalCode,
+        hideIcon,
+        iconStyle,
+        disabled,
+        postalCode,
+        onChange,
+        onReady,
+        onFocus,
+        onBlur,
+    } = props;
+    const stripe = useStripe();
+    const elements = useElements();
+    const cardElementId = useUniqueIdentifier('card-element');
+    const theme = useTheme().rainbow;
+    const cardElementOptions = useMemo(
+        () => ({
+            style: {
+                base: {
+                    iconColor: theme.palette.text.main,
+                    fontFamily: '"Lato", Arial, sans-serif',
+                    backgroundColor: theme.palette.background.main,
+                    color: theme.palette.text.main,
+                    fontSize: '1.1rem',
+                    '::placeholder': {
+                        color: theme.palette.text.header,
+                        fontVariant: 300,
+                    },
+                },
+            },
+            hidePostalCode,
+            hideIcon,
+            iconStyle,
+            disabled,
+            value: { postalCode },
+        }),
+        [theme, hidePostalCode, hideIcon, iconStyle, disabled, postalCode],
+    );
+    useEffect(() => {
+        if (!elements) {
+            return;
+        }
+        const cardElement = elements.getElement(CardElement);
+        if (cardElement) {
+            cardElement.update({
+                value: { postalCode },
+            });
+        }
+    }, [elements, postalCode]);
+
+    const handleChange = event => {
+        if (!stripe || !elements) {
```

Comment:
i"m not sure if we are being over defensive here, I guess that onChange is not going to be fired if the elements and stripe are not ready, in fact, the component shouldn't render if those values are not ready. But if I'm wrong ... make the if condition different

```js
if (stripe && elements) {
...
}
```